lib/location/method_gnss.c

```diff
@@ -365,6 +379,13 @@ static void method_gnss_request_assistance(void)
 		return;
 	}
 
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_PGPS)) {
+		/* ephemerides come from P-GPS; almanacs not desired in this configuration */
+		pgps_agps_request.sv_mask_ephe = agps_request.sv_mask_ephe;
+		agps_request.sv_mask_ephe = 0;
```

Comment:
I tested your code and noticed that this thing does not work correctly with filtered ephemeris feature - A-GPS will fetch the ephemerides for the filtered subset anyway. But this minor issue aside, I am happy you are working on location lib. I am aware there are some issues with how A-GPS and P-GPS are used. There is even ticket for that but I haven't got around starting work on it https://projecttools.nordicsemi.no/jira/browse/NCSDK-14060

Reply:
Thanks for finding this.  I confirmed it just now with ATv2 as well.

I have opened a ticket with the author of the cloud side to respond to (IRIS-4158).